manifest/manifestlist/manifestlist_test.go

```diff
@@ -109,3 +110,105 @@ func TestManifestList(t *testing.T) {
 		}
 	}
 }
+
+// TODO (mikebrow): add annotations on the index and individual manifests
+var expectedOCIImageIndexSerialization = []byte(`{
```

Comment:
platform is not a required attribute on OCI indexes.

I am not sure about the idea of replacing tags with an index fetch. We have definitely discussed but the value may not outweigh the cost of change. Let's remember that `org.opencontainers.image.ref.name` is only valid on `index.json` on an "Image Layout" and registries do not serve image layouts, they serve individual components.